simple_graph.py

```diff
@@ -93,3 +95,158 @@ def adjacent(self, n1, n2):
                 return True
```

Comment:
This is going to be a pretty slow operation for a reasonably large and complete graph.  Can you come up with a faster way to do it?